client/state/data-layer/wpcom/login-2fa/index.js

```diff
@@ -20,71 +14,91 @@ import {
 	getTwoFactorRememberMe,
 	getTwoFactorPushPollInProgress
 } from 'state/login/selectors';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { rawHttp } from 'state/data-layer/http/actions';
 
 /***
  * Module constants
  */
 const POLL_APP_PUSH_INTERVAL_SECONDS = 5;
 
 /***
- * Checks the status of the push notification auth
+ * Dispatch action to perform http request
  *
- * @param {Object}   store  Global redux store
- * @returns {Promise}		Promise of result from the API
+ * @param {Object}	store  Global redux store
+ * @param {Object}	action dispathced action
  */
-const doAppPushRequest = ( store ) => {
-	return request.post( 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint' )
-		.withCredentials()
-		.set( 'Content-Type', 'application/x-www-form-urlencoded' )
-		.accept( 'application/json' )
-		.send( {
+const requestTwoFactorPushNotificationStatus = ( store, action ) => {
+	store.dispatch( rawHttp( {
+		url: 'https://wordpress.com/wp-login.php?action=two-step-authentication-endpoint',
```

Comment:
I debated with this a long time myself =)
`http` "mirrors" `wpcom-http` so imo it shouldn't go there
`third-party` is for, well, third parties, and this is not a third party API, it's just different from wpcom rest API

So I dunno really, open for suggestions. It shouldn't be a blocker for this PR though...